sotodlib/tod_ops/jumpfind.py

```diff
@@ -0,0 +1,406 @@
+import numpy as np
+import scipy.signal as sig
+import scipy.ndimage as simg
+from so3g.proj import RangesMatrix
+from skimage.restoration import denoise_tv_chambolle
+
+
+def _jumpfind(x, min_chunk, min_size, win_size, max_depth=-1, depth=0, **kwargs):
+    """
+    Recursive edge detection based jumpfinder.
+
+    Note that the jumpfinder is very sensitive to changes in parameters
+    and the parameters are not independant of each other,
+    so it may take some playing around to get it to work properly.
+
+    Arguments:
+
+        x: Data to jumpfind on, expects 1D
+           Note that x will by modified in place with jumps removed
+           pass x.copy() to preserve x (say to pass to a better jump fixer)
+
+        min_chunk: The smallest chunk of data to look for jumps in
+
+        min_size: The smalled jump size counted as a jump
+
+        win_size: Number of samples to average over when checking jump size
+
+        max_depth: The maximum recursion depth, set negetive to not use
+
+        depth: The current recursion depth
+
+        **kwargs: Arguments to pass to scipy.signal.find_peaks
+
+    Returns:
+
+        jumps: The indices of jumps in x
+               There is some uncertainty on order of 1 sample
+               Jumps within min_chunk of each other may not be distinguished
+    """
+    if len(x) < min_chunk:
+        return np.array([], dtype=int)
+
+    # If std is basically 0 no need to check for jumps
+    if np.isclose(x.std(), 0.0):
+        return np.array([], dtype=int)
+
+    # Mean subtract the data
+    _x = x - x.mean()
+
+    # Scale data to have std of order 1
+    if _x.std() > 10:
+        _x = _x / (10 ** (int(np.log10(_x.std()))))
```

Comment:
It's because having several large jumps can skew the std high. There are tod's where the std in the not so jumpy regions are <1 but the tod as a whole is something closer to 5 or 6. I actually don't know if this is needed. When I did some testing where I rescaled things to have higher noise level (ie: white noise with std of 50 instead of 1) this helped find some jumps that were below the noise level. For real data I don't know if we need this since we shouldn't have anything with stds that high (I hope).

Reply:
Rescaling by the stdev makes sense.  My point was that snapping to powers of 10 is not doing anything other than potentially discretizing the behavior.  Why not just:
```
if _x.std() != 0:
  _x /= _x.std()
```